src/modules/user/user.service.ts

```diff
@@ -0,0 +1,38 @@
+import {
+  Inject,
+  Injectable,
+  NotFoundException,
+  UnauthorizedException,
+} from '@nestjs/common';
+import { UserRepository } from './user.repository';
+import { User } from '@prisma/client';
+
+@Injectable()
+export class UserService {
+  constructor(
+    @Inject('UserRepository') private readonly userRepository: UserRepository,
+  ) {}
+
+  async login(email: string, password: string): Promise<User | null> {
+    const user = await this.userRepository.findByEmail(email);
+
+    if (!user) {
+      throw new NotFoundException(`User with username ${email} not found`);
+    }
+    const isPasswordValid = user.password === password;
+
+    if (!isPasswordValid) {
+      throw new UnauthorizedException('Invalid password');
+    }
+
+    return user;
+  }
+
+  async updatePassword(id: number, newPassword: string): Promise<User | null> {
+    const user = await this.userRepository.updatePassword(id, newPassword);
```

Comment:
The repository should be responsible for throwing this error, now imagine we have to update the user in 3 different controllers, this results in 3x repeating this if statement. If it's the repository that takes care of it, only the repository will throw without any code repetition. Semantically it also makes more sense for the repo to throw.